src/streamsync/ui_manager.py

```diff
@@ -115,9 +117,57 @@ def _prepare_value(value):
     return str(value)
 
 
-def _create_component(component_tree: ComponentTree,  component_type: str, **kwargs) -> Component:
+def _check_parent_child_relations(
+        component_tree: ComponentTree,
+        parent_id: str,
+        component_type: str
+):
+    # Import required inside a function:
+    # StreamsyncUIManager class stores actual type maps,
+    # but importing it directly causes a circular import
+    from streamsync.ui import StreamsyncUIManager
+
+    if component_type == "root":
+        raise UIError("Root component cannot be a child component.")
```

Comment:
Root check is required to avoid the case when 'root' type can "slip" through a wildcard (`*`).